FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+
+static void prvInitializeCommandContext( CommandContext_t * pxContext )
+{
+    pxContext->complete = false;
+    pxContext->pResponseQueue = NULL;
+    pxContext->returnStatus = MQTTSuccess;
+    pxContext->pPublishInfo = NULL;
+    pxContext->pSubscribeInfo = NULL;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvDestroyCommandContext( CommandContext_t * pxContext )
+{
+    ( void ) pxContext;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvAddAck( uint16_t usPacketId,
+                       CommandContext_t * pxContext,
+                       CommandCallback_t xCallback )
+{
+    int32_t i = 0;
+
+    for( i = 0; i < PENDING_ACKS_MAX_SIZE; i++ )
+    {
+        if( pxPendingAcks[ i ].packetId == MQTT_PACKET_ID_INVALID )
+        {
+            pxPendingAcks[ i ].packetId = usPacketId;
+            pxPendingAcks[ i ].pCommandContext = pxContext;
+            pxPendingAcks[ i ].callback = xCallback;
+            break;
+        }
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static AckInfo_t prvPopAck( uint16_t usPacketId )
+{
+    int32_t i = 0;
+    AckInfo_t xFoundAck = { 0 };
+
+    for( i = 0; i < PENDING_ACKS_MAX_SIZE; i++ )
+    {
+        if( pxPendingAcks[ i ].packetId == usPacketId )
+        {
+            xFoundAck = pxPendingAcks[ i ];
+            pxPendingAcks[ i ].packetId = MQTT_PACKET_ID_INVALID;
+            pxPendingAcks[ i ].pCommandContext = NULL;
+            pxPendingAcks[ i ].callback = NULL;
+            break;
+        }
+    }
+
+    return xFoundAck;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvAddSubscription( const char * pTopicFilter,
+                                uint16_t topicFilterLength,
+                                QueueHandle_t pxQueue )
+{
+    int32_t i = 0;
+
+    for( i = 0; i < SUBSCRIPTIONS_MAX_COUNT; i++ )
+    {
+        if( pxSubscriptions[ i ].topicFilterLength == 0 )
+        {
+            pxSubscriptions[ i ].topicFilterLength = topicFilterLength;
+            pxSubscriptions[ i ].pResponseQueue = pxQueue;
+            memcpy( pxSubscriptions[ i ].pTopicFilter, pTopicFilter, topicFilterLength );
+            break;
+        }
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvRemoveSubscription( const char * pTopicFilter,
+                                   size_t topicFilterLength,
+                                   QueueHandle_t pxQueue )
+{
+    /* TODO: Unused for now, but can be used to remove a single subscription
+     * when multiple apps are subscribed to the same topic. */
+    ( void ) pxQueue;
+    int32_t i = 0;
+
+    for( i = 0; i < SUBSCRIPTIONS_MAX_COUNT; i++ )
+    {
+        if( pxSubscriptions[ i ].topicFilterLength == topicFilterLength )
+        {
+            if( ( strncmp( pxSubscriptions[ i ].pTopicFilter, pTopicFilter, topicFilterLength ) == 0 ) && true )
```

Comment:
why is ` && true` added here?

Reply:
This is an artifact of a previous iteration of the function that had to do with the TODO of removing only a signle subscription. It originally checked that pxQueue matched as well. I will remove the TODO and the extraneous `true`